File: scripts/statusline_setup.py
#!/usr/bin/env python3
import json
import os
from pathlib import Path

PLUGIN_NAME = "cc-distribution"


def _settings_path(root: Path) -> Path:
    return root / ".claude" / "settings.json"


def _load_settings(path: Path) -> dict:
    if not path.exists():
        return {}
    try:
        with path.open("r", encoding="utf-8") as handle:
            return json.load(handle)
    except json.JSONDecodeError:
        return {}


def _save_settings(path: Path, settings: dict) -> None:
    path.parent.mkdir(parents=True, exist_ok=True)
    with path.open("w", encoding="utf-8") as handle:
        json.dump(settings, handle, indent=2, sort_keys=True)
        handle.write("\n")


def _statusline_command() -> str:
    return (
        "bash -lc '"
        "plugin_dir=$(ls -td ~/.claude/plugins/cache/" + PLUGIN_NAME + "/" + PLUGIN_NAME + "/*/ 2>/dev/null | head -1);"
        "if [ -z \"$plugin_dir\" ]; then exit 0; fi;"
        "python3 \"${plugin_dir}scripts/statusline.py\"'"
    )


def main() -> int:
    root = Path(os.environ.get("CLAUDE_PROJECT_DIR", os.getcwd())).resolve()
    settings_path = _settings_path(root)
    settings = _load_settings(settings_path)

    if "statusLine" in settings:
        return 0

    settings["statusLine"] = {
        "type": "command",
        "command": _statusline_command(),
    }
    _save_settings(settings_path, settings)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: scripts/statusline_setup.js
#!/usr/bin/env node
const fs = require("fs");
const path = require("path");

const PLUGIN_NAME = "cc-distribution";

const projectDir = process.env.CLAUDE_PROJECT_DIR || process.cwd();
const settingsPath = path.join(projectDir, ".claude", "settings.json");

const readSettings = () => {
  try {
    return JSON.parse(fs.readFileSync(settingsPath, "utf8"));
  } catch (err) {
    return {};
  }
};

const writeSettings = (settings) => {
  fs.mkdirSync(path.dirname(settingsPath), { recursive: true });
  fs.writeFileSync(settingsPath, JSON.stringify(settings, null, 2) + "\n");
};

const statuslineCommand = () =>
  "bash -lc '" +
  "plugin_dir=$(ls -td ~/.claude/plugins/cache/" +
  PLUGIN_NAME +
  "/" +
  PLUGIN_NAME +
  "/*/ 2>/dev/null | head -1);" +
  "if [ -z \"$plugin_dir\" ]; then exit 0; fi;" +
  "node \"${plugin_dir}scripts/statusline.js\"'";

const settings = readSettings();
if (!settings.statusLine) {
  settings.statusLine = {
    type: "command",
    command: statuslineCommand(),
  };
  writeSettings(settings);
}
